sql/core/src/main/scala/org/apache/spark/sql/execution/DataSourceScanExec.scala

```diff
@@ -141,7 +138,8 @@ case class RowDataSourceScanExec(
   // Don't care about `rdd` and `tableIdentifier` when canonicalizing.
   override def doCanonicalize(): SparkPlan =
     copy(
-      fullOutput.map(QueryPlan.normalizeExpressions(_, fullOutput)),
+      // Only the required column names matter when checking equality.
+      output.map(a => a.withExprId(ExprId(-1))),
```

Comment:
Why does this not use `normalizeExpressions`? It seems odd to use a special case fixed ID here.